packages/aws-cdk-lib/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -316,12 +346,27 @@ export class LustreFileSystem extends FileSystemBase {
     this.validateAutomaticBackupRetention(deploymentType, lustreConfiguration.automaticBackupRetention);
 
     this.validateDailyAutomaticBackupStartTime(lustreConfiguration.automaticBackupRetention, lustreConfiguration.dailyAutomaticBackupStartTime);
+    this.validateFiileSystemTypeVersion(deploymentType, props.fileSystemTypeVersion);
   }
 
   /**
-   * Validates the auto import policy
+   * Validates the file system type version
    */
+  private validateFiileSystemTypeVersion(deploymentType: LustreDeploymentType, fileSystemTypeVersion?: FileSystemTypeVersion): void {
+    if (fileSystemTypeVersion === undefined) {
+      return;
+    }
 
+    if (fileSystemTypeVersion === FileSystemTypeVersion.V_2_10) {
+      if (!deploymentType.startsWith('SCRATCH') && deploymentType !== LustreDeploymentType.PERSISTENT_1) {
+        throw new Error('fileSystemTypeVersion V_2_10 is only supported for SCRATCH and PERSISTENT_1 deployment types');
+      }
```

Comment:
can we also validate 
```


2.12 is supported by all Lustre deployment types, except for PERSISTENT_2 with a metadata configuration mode.

```

Reply:
As I described [above](https://github.com/aws/aws-cdk/pull/31136#issue-2471729944), `LustreFileSystemProps` does not have metadata configuration prop and I think it is difficult to add validation for that.

What do you think about this?

Reply:
My bad. I totally missed the note in the PR description. Since this prop doesn't exists yet, I think it's fine to skip this validation now.

Reply:
But can you add a TODO comment to indicate that we should add one once we support metadata config?

Reply:
I've added TODO comment. Could you please confirm?